server/src/test/java/org/spine3/server/entity/FieldMasksShould.java

```diff
@@ -0,0 +1,172 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.entity;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.FieldMask;
+import org.junit.Test;
+import org.spine3.protobuf.TypeUrl;
+import org.spine3.test.aggregate.Project;
+import org.spine3.test.aggregate.ProjectId;
+import org.spine3.test.aggregate.Task;
+import org.spine3.test.aggregate.TaskId;
+import org.spine3.test.commandservice.customer.Customer;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+import static org.spine3.test.Tests.assertMatchesMask;
+import static org.spine3.test.Tests.hasPrivateUtilityConstructor;
+import static org.spine3.test.Verify.assertSize;
+
+/**
+ * @author Dmytro Dashenkov
+ */
+public class FieldMasksShould {
+
+    @Test
+    public void have_private_constructor() {
+        assertTrue(hasPrivateUtilityConstructor(FieldMasks.class));
+    }
+
+    @Test
+    public void apply_mask_to_single_message() {
+        final FieldMask fieldMask = Given.fieldMask(Project.ID_FIELD_NUMBER, Project.NAME_FIELD_NUMBER);
+
+        final Project original = Given.newProject("some-string-id");
+
+        final Project masked = FieldMasks.applyMask(fieldMask, original, Given.TYPE);
+
+        assertNotEquals(original, masked);
+
+        assertMatchesMask(masked, fieldMask);
+    }
+
+    @SuppressWarnings("MethodWithMultipleLoops")
+    @Test
+    public void apply_mask_to_message_collections() {
+        final FieldMask fieldMask = Given.fieldMask(Project.STATUS_FIELD_NUMBER, Project.TASK_FIELD_NUMBER);
+        final int count = 5;
+
+        final Collection<Project> original = new LinkedList<>();
+
+        for (int i = 0; i < count; i++) {
+            final Project project = Given.newProject(String.format("project-%s", i));
+            original.add(project);
+        }
+
+        final Collection<Project> masked = FieldMasks.applyMask(fieldMask, original, Given.TYPE);
+
+        assertSize(original.size(), masked);
+
+        // Collection references are not the same
+        // noinspection ObjectEquality
+        assertFalse(original == masked);
```

Comment:
If you use `noinspection` you should remember two things:

* That's just for IDEA. All code inspection tools use `@SuppressWarnings`.
* You should explain why it is fine to suppress warnings in this case.

Could you please review the contents of the files in this PR and check to use annotation-based approach and have proper comments next to the suppression.